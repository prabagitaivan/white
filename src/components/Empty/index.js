import React, { memo } from 'react'
import { makeStyles } from '@material-ui/styles'

const useStyles = makeStyles(
  {
    root: {
      textAlign: 'center',
      marginTop: '40vh'
    }
  },
  { name: 'Empty' }
)

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Still Empty</div>
      <span role='img' aria-label='three-wise-monkeys'>
        ğŸ™ˆğŸ™‰ğŸ™Š
      </span>
    </div>
  )
})
