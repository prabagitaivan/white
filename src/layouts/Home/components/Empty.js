import React, { PureComponent } from 'react'
import { empty as styles } from '../styles'

export default class Empty extends PureComponent {
  render () {
    return (
      <div style={styles.root}>
        <div>Still Empty</div>
        <span role='img' aria-label='three-wise-monkeys'>
          ðŸ™ˆðŸ™‰ðŸ™Š
        </span>
      </div>
    )
  }
}
