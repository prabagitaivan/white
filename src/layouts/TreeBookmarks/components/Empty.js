import React, { memo } from 'react'
import { empty as useStyles } from '../styles'

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Still Empty</div>
      <span role='img' aria-label='three-wise-monkeys'>
        ☁️🌧️⛈️
      </span>
    </div>
  )
})
