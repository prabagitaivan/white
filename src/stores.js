import { createStore as ReduxCreateStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'
import reducers from './reducers'
import sagas from './sagas'

export const createStore = preloadedState => {
  const sagaMiddleware = createSagaMiddleware()
  const enhancer = applyMiddleware(sagaMiddleware)

  const store = ReduxCreateStore(reducers, preloadedState, enhancer)
  sagas.forEach(sagaMiddleware.run)

  return store
}

export default createStore()
