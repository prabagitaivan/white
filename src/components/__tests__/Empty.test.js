import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from '../../stores'
import Styles from '../../styles'
import Empty from '../Empty'

describe('components Empty', () => {
  describe('snapshots', () => {
    it('return correct snapshot', () => {
      const { container } = render(
        <Provider store={createStore()}>
          <Styles>
            <Empty />
          </Styles>
        </Provider>
      )
      expect(container.firstChild).toMatchSnapshot()
    })
  })
})
