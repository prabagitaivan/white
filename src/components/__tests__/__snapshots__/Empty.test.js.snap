// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Empty snapshots return correct snapshot 1`] = `
<div
  class="WhiteEmpty-root"
>
  <div>
    Still Empty
  </div>
  <span
    aria-label="three-wise-monkeys"
    role="img"
  >
    ðŸ™ˆðŸ™‰ðŸ™Š
  </span>
</div>
`;
